internal/logger/targets.go

```diff
@@ -148,8 +146,76 @@ func UpdateTargets(cfg Config) error {
 		}
 	}
 	atomic.StoreInt32(&nTargets, int32(len(updated)))
-	cancelAllTargets() // cancel running targets
-	targets = updated
+	cancelAllSystemTargets() // cancel running targets
+	systemTargets = updated
+	swapMu.Unlock()
+	return nil
+}
+
+func cancelAuditWebhookTargets() {
+	for _, tgt := range auditTargets {
+		if tgt.Endpoint() != kafka.NAME {
+			tgt.Cancel()
+		}
```

Comment:
This is a bit hacky, and a trap for the future.

Add a `Type() TargetType` that allow you to get the exact underlying type, and don't duplicate code:

```
type TargetType uint8

func existingTargets(t TargetType) []Target {
	awTgts := make([]Target{}, 0, len(auditTargets))
	for _, tgt := range auditTargets {
		if tgt.Type() == t {
			awTgts = append(awTgts, tgt)
		}
	}
	return awTgts
}

func cancelTargetType(t TargetType) {
	for _, tgt := range auditTargets {
		if tgt.Type() ==t {
			tgt.Cancel()
		}
	}
}

```

Reply:
Done.